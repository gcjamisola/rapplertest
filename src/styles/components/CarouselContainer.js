import styled, { injectGlobal } from 'styled-components';
import * as utils from '../utils';

const { media } = utils;

const CarouselContainer = styled.div`
    background: #419be0;
    color: #333;
    height: 475px;
    margin: 0 auto;
    padding: 40px;

    ${media.desktop`width: 88%;`}
    ${media.tablet`
        height: 350px;
        width: 100%;
    `}
    ${media.phone`height: 300px;`}
`;

const StyledSlider = styled.div`
    .slick-initialized.slick-slider {
        height: 450px;
    }

    ${media.tablet`height: 400px;`}
    ${media.phablet`height: 375px;`}
    ${media.phone`height: 300px`}
`;

const SliderElement = styled.div`
    background-color: rgba(255, 255, 255, .25);
    color: black;
    height: 150px;
    position: relative;
    top: -10em;
    width: 100%;

    ${media.phablet`
        height: 100px;
        top: -7em;
    `}
`;

const SliderImage = styled.img`
    height: 450px;
    width: 100%;

    ${media.desktop`height: 400px;`}
    ${media.tablet`height: 350px`}
    ${media.phablet`height: 300px`}
`;

const SliderText = styled.div`
    color: #000;
    left: 1em;
    position: relative;
    width: 95%;
`;

const SliderTitle = SliderText.extend`
    font-size: 1.5em;
    font-weight: bold;
    margin: 0;
    top: 1em;

    ${media.phablet`
        font-size: 1.25em;
    `}
`;

const SliderDescription = SliderText.extend`
    bottom: -5em;
    margin-left: .5em;

    ${media.phablet`
        display: none;
    `}
`;

const ReadMore = styled.div`
    bottom: 1em;
    color: rgba(44, 44, 44, 0.75);
    display: none;
    font-weight: bold;
    position: absolute;
    right: 1em;

    ${media.phablet`
        display: block;
    `}
`;

export { 
    CarouselContainer, 
    ReadMore,
    StyledSlider,
    SliderElement, 
    SliderImage, 
    SliderTitle, 
    SliderDescription
 };