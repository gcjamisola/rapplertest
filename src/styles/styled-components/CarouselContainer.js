import styled from 'styled-components';

const sizes = {
    desktop: 992,
    tablet: 768,
    phone: 376
}

const media = Object.keys(sizes).reduce((acc, label) => {
    acc[label] = (...args) => css`
      @media (max-width: ${sizes[label] / 16}em) {
        ${css(...args)}
      }
    `
    return acc
  }, {})

const StyledSlider = styled.div`
    .slick-initialized.slick-slider {
        height: 550px;
    }
`;

const CarouselContainer = styled.div`
    background: #419be0;
    color: #333;
    height: 575px;
    margin: 0 auto;
    padding: 40px;
    width: 100%;
`;

const SliderElement = styled.div`
    background-color: rgba(255, 255, 255, .25);
    color: black;
    height: 150px;
    position: relative;
    top: -10em;
    width: 100%;
`;

const SliderImage = styled.img`
    height: 550px;
    width: 100%;
`;

const SliderText = styled.div`
    color: #000;
    left: 1em;
    position: relative;
    width: 95%;
`;

const SliderTitle = SliderText.extend`
    font-size: 1.5em;
    font-weight: bold;
    margin: 0;
    top: 1em;
`;

const SliderDescription = SliderText.extend`
    bottom: -5em;
    margin-left: .5em;
`;

export { 
    CarouselContainer, 
    StyledSlider, 
    SliderElement, 
    SliderImage, 
    SliderTitle, 
    SliderDescription
 };