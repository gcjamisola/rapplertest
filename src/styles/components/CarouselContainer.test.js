import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import * as scCarousel from './CarouselContainer';

test('it works', () => {
  let tree = renderer.create(<scCarousel.CarouselContainer />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.ReadMore />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.StyledSlider />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.SliderElement />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.SliderImage />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.SliderTitle />).toJSON()
	expect(tree).toMatchSnapshot()

	tree = renderer.create(<scCarousel.SliderDescription />).toJSON()
  expect(tree).toMatchSnapshot()
})
