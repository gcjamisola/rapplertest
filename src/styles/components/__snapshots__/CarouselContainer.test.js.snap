// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  background: #419be0;
  color: #333;
  height: 475px;
  margin: 0 auto;
  padding: 40px;
}

@media (max-width:992px) {
  .c0 {
    width: 88%;
  }
}

@media (max-width:768px) {
  .c0 {
    height: 350px;
    width: 100%;
  }
}

@media (max-width:376px) {
  .c0 {
    height: 300px;
  }
}

<div
  className="c0"
/>
`;

exports[`it works 2`] = `
.c0 {
  bottom: 1em;
  color: rgba(44,44,44,0.75);
  display: none;
  font-weight: bold;
  position: absolute;
  right: 1em;
}

@media (max-width:572px) {
  .c0 {
    display: block;
  }
}

<div
  className="c0"
/>
`;

exports[`it works 3`] = `
.c0 .slick-initialized.slick-slider {
  height: 450px;
}

@media (max-width:768px) {
  .c0 {
    height: 400px;
  }
}

@media (max-width:572px) {
  .c0 {
    height: 375px;
  }
}

@media (max-width:376px) {
  .c0 {
    height: 300px;
  }
}

<div
  className="c0"
/>
`;

exports[`it works 4`] = `
.c0 {
  background-color: rgba(255,255,255,.25);
  color: black;
  height: 150px;
  position: relative;
  top: -10em;
  width: 100%;
}

@media (max-width:572px) {
  .c0 {
    height: 100px;
    top: -7em;
  }
}

<div
  className="c0"
/>
`;

exports[`it works 5`] = `
.c0 {
  height: 450px;
  width: 100%;
}

@media (max-width:992px) {
  .c0 {
    height: 400px;
  }
}

@media (max-width:768px) {
  .c0 {
    height: 350px;
  }
}

@media (max-width:572px) {
  .c0 {
    height: 300px;
  }
}

<img
  className="c0"
/>
`;

exports[`it works 6`] = `
.c0 {
  color: #000;
  left: 1em;
  position: relative;
  width: 95%;
  font-size: 1.5em;
  font-weight: bold;
  margin: 0;
  top: 1em;
}

@media (max-width:572px) {
  .c0 {
    font-size: 1.25em;
  }
}

<div
  className="c0"
/>
`;

exports[`it works 7`] = `
.c0 {
  color: #000;
  left: 1em;
  position: relative;
  width: 95%;
  bottom: -5em;
  margin-left: .5em;
}

@media (max-width:572px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;
